overlord/servicestate/servicestate.go

```diff
@@ -127,7 +127,20 @@ func serviceControlTs(st *state.State, appInfos []*snap.AppInfo, inst *Instructi
 		sort.Strings(explicitSvcs)
 		cmd.ExplicitServices = explicitSvcs
 
-		summary := fmt.Sprintf("Run service command %q for services %q of snap %q", cmd.Action, svcs, cmd.SnapName)
+		// When composing the task summary, prefer using the explicit
+		// services, if that's not empty
+		var summary string
+		if len(explicitSvcs) > 0 {
```

Comment:
I wonder if a switch would make this easier to read?
```
switch {
		case len(explicitSvcs) > 0:
			// When composing the task summary, prefer using the explicit
			// services, if that's not empty
			summary = fmt.Sprintf("Run service command %q for services %q of snap %q", cmd.Action, explicitSvcs, cmd.SnapName)
		case inst.Action == "restart":
			// Use a generic message, since we cannot know the exact list of
			// services affected
			summary = fmt.Sprintf("Run service command %q for running services of snap %q", cmd.Action, cmd.SnapName)
		default:
			// XXX: what to put here?
			summary = fmt.Sprintf("Run service command %q for services %q of snap %q", cmd.Action, svcs, cmd.SnapName)
		}
```
Feels all a bit more explicit to me but maybe just a personal taste.

Reply:
Mmm... I don't have anything against switches, but in this case it would force me to repeat the text `"Run service command %q for services %q of snap %q"` twice, so I'm not sure I like it more...